pkg/controller/atlasauditing/controller.go

```diff
@@ -0,0 +1,115 @@
+package atlasauditing
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translayer/auditing"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1alpha1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/customresource"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/watch"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
+)
+
+type AuditingStatus string
+
+// AtlasAuditingReconciler reconciles an AtlasAuditing object
+type AtlasAuditingReconciler struct {
+	watch.DeprecatedResourceWatcher
+	Client          client.Client
+	Log             *zap.SugaredLogger
+	Scheme          *runtime.Scheme
+	EventRecorder   record.EventRecorder
+	auditingService auditing.Service
+}
+
+func (r *AtlasAuditingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// TODO: make sure this is done when constructing AtlasAuditingReconciler
+	// log := r.Log.With("atlasdatabaseuser", req.NamespacedName)
+
+	auditing := &v1alpha1.AtlasAuditing{}
+	result := customresource.PrepareResource(ctx, r.Client, req, auditing, r.Log)
+	if !result.IsOk() {
+		return result.ReconcileResult(), nil
+	}
+
+	if customresource.ReconciliationShouldBeSkipped(auditing) {
+		r.Log.Infow(fmt.Sprintf("-> Skipping AtlasAuditing reconciliation as annotation %s=%s", customresource.ReconciliationPolicyAnnotation, customresource.ReconciliationPolicySkip), "spec", auditing.Spec)
+		return workflow.OK().ReconcileResult(), nil
+	}
+
+	// TODO: Fix! we no longer want the status to flip&flap but be updated only at the end
+	//       not sure what we chose to replace this with
+	// workflowCtx := customresource.MarkReconciliationStarted(r.Client, databaseUser, log, ctx)
```

Comment:
see https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1595 with a replacement, we have everything in master already.